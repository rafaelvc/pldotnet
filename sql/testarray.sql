CREATE OR REPLACE FUNCTION sumArray(a integer[]) RETURNS integer AS $$
return a[0] + a[1] + a[2];
$$ LANGUAGE pldotnet;
SELECT sumArray( ARRAY[4,1,5] ) = integer '10';

