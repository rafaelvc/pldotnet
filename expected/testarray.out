CREATE OR REPLACE FUNCTION sumArrayInt(a integer[]) RETURNS integer AS $$
return a[0] + a[1] + a[2];
$$ LANGUAGE plcsharp;
SELECT sumArrayInt( ARRAY[4,1,5] ) = integer '10';
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION sumArrayNum(a numeric[]) RETURNS numeric AS $$
return a[0] + a[1] + a[2];
$$ LANGUAGE plcsharp;
SELECT sumArrayNum( ARRAY[1.00002, 1.00003, 1.00004] ) = numeric '3.00009';
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION sumArrayText(a text[]) RETURNS text AS $$
return a[0] + a[1] + a[2] + a[3];
$$ LANGUAGE plcsharp;
SELECT sumArrayText( ARRAY['Rafael', ' da', ' Veiga', ' Cabral'] ) = varchar 'Rafael da Veiga Cabral';
 ?column? 
----------
 t
(1 row)

